libsql-server/src/auth/user_auth_strategies/mod.rs

```diff
@@ -2,17 +2,26 @@ pub mod disabled;
 pub mod http_basic;
 pub mod jwt;
 
-use axum::http::HeaderValue;
+use anyhow::Context;
 pub use disabled::*;
 pub use http_basic::*;
 pub use jwt::*;
 
 use super::{AuthError, Authenticated};
 
 pub struct UserAuthContext {
-    pub user_credential: Option<HeaderValue>,
+    pub scheme: Option<String>,
+    pub token: Option<String>, // token might not be required in some cases
```

Comment:
Since we are converting things via TryFrom now it would be a good time to turn these into private fields and using setters/gettings. It will make refactoring easier in the future. We should in general avoid pub fields, something that we have a lot of now but plan on fixing. Especially, if we are going to release this as a crate.